### PR TITLE
MINOR: update docs with regard to improved resilience of Kafka Streams

### PR DESCRIPTION
**Pull request by mjsax**

### CODE REVIEW

**Comment by mjsax** [comment]
Call for review @joel-hamill @guozhangwang @bbejeck @dguy

File: docs/upgrade.html [hunk 1]
```diff
@@ -124,6 +124,11 @@ <h5><a id="upgrade_101_notable" href="#upgrade_101_notable">Notable changes in 1
 <ul>
     <li>Restored binary compatibility of AdminClient's Options classes (e.g. CreateTopicsOptions, DeleteTopicsOptions, etc.) with
         0.11.0.x. Binary (but not source) compatibility had been broken inadvertently in 1.0.0.</li>
+    <li>Kafka Streams is more robust against broker communication errors. Instead of stopping the Kafka Streams client with a fatal exception,
+	Kafka Streams tries to self-heal and reconnect to the cluster. Using the new <code>AdminClient</code> you have also more control how often
+	Kafka Streams should retry and configure timeouts fine grained (instead of hard coded retries as in older version).</li>
```
Comment (bbejeck) [inline docs/upgrade.html]: nit: "fine grained" -> fine-grained

Comment (guozhangwang) [reply docs/upgrade.html]: The new `AdminClient` is only added starting 1.1.0 right?

Comment (mjsax) [reply docs/upgrade.html]: Ups. Wrong section.

**Comment by mjsax** [comment]
Updated this.

File: docs/upgrade.html [hunk 2]
```diff
@@ -66,6 +66,10 @@ <h5><a id="upgrade_110_notable" href="#upgrade_110_notable">Notable changes in 1
     <li>The kafka artifact in Maven no longer depends on log4j or slf4j-log4j12. Similarly to the kafka-clients artifact, users
         can now choose the logging back-end by including the appropriate slf4j module (slf4j-log4j12, logback, etc.). The release
         tarball still includes log4j and slf4j-log4j12.</li>
+    <li>Kafka Streams is more robust against broker communication errors. Instead of stopping the Kafka Streams client with a fatal exception,
+	Kafka Streams tries to self-heal and reconnect to the cluster. Using the new <code>AdminClient</code> you have also more control how often
+	Kafka Streams should retry and configure timeouts fine-grained (instead of hard coded retries as in older version).</li>
+    <li>Kafka Streams rebalance time was reduced further making Kafka Streams more responsive.</li>
```
Comment (guozhangwang) [inline docs/upgrade.html]: nit: could we add a reference link to `doc/streams/upgrade.html` for users to read how to set the configs etc?

**Review by guozhangwang — COMMENTED** [review]
One minor comment, otherwise LGTM

File: docs/upgrade.html [hunk 3]
```diff
@@ -66,6 +66,10 @@ <h5><a id="upgrade_110_notable" href="#upgrade_110_notable">Notable changes in 1
     <li>The kafka artifact in Maven no longer depends on log4j or slf4j-log4j12. Similarly to the kafka-clients artifact, users
         can now choose the logging back-end by including the appropriate slf4j module (slf4j-log4j12, logback, etc.). The release
         tarball still includes log4j and slf4j-log4j12.</li>
+    <li>Kafka Streams is more robust against broker communication errors. Instead of stopping the Kafka Streams client with a fatal exception,
+	Kafka Streams tries to self-heal and reconnect to the cluster. Using the new <code>AdminClient</code> you have also more control how often
+	Kafka Streams should retry and configure timeouts fine-grained (instead of hard coded retries as in older version).</li>
```
Comment (joel-hamill) [inline docs/upgrade.html]: suggest `...hard-coded retries...`

**Review by joel-hamill — APPROVED** [review]

**Comment by mjsax** [comment]
Updates this. Will merge after build passed (of there is no objection raised).

**Review by bbejeck — APPROVED** [review]
thanks @mjsax, LGTM

**Comment by mjsax** [comment]
Retest this please.

**Comment by mjsax** [comment]
Retest this please.

**Comment by mjsax** [comment]
Retest this please.

**Comment by mjsax** [comment]
Retest this please.

File: docs/upgrade.html [hunk 4]
```diff
@@ -67,6 +67,11 @@ <h5><a id="upgrade_110_notable" href="#upgrade_110_notable">Notable changes in 1
         can now choose the logging back-end by including the appropriate slf4j module (slf4j-log4j12, logback, etc.). The release
         tarball still includes log4j and slf4j-log4j12.</li>
     <li><a href="https://cwiki.apache.org/confluence/x/uaBzB">KIP-225</a> changed the metric "records.lag" to use tags for topic and partition. The original version with the name format "{topic}-{partition}.records-lag" is deprecated and will be removed in 2.0.0.</li>
+    <li>Kafka Streams is more robust against broker communication errors. Instead of stopping the Kafka Streams client with a fatal exception,
+	Kafka Streams tries to self-heal and reconnect to the cluster. Using the new <code>AdminClient</code> you have also more control how often
```
Comment (dguy) [inline docs/upgrade.html]: not sure this sentence sounds correct.
the part `you have also more control how often Kafka Streams should retry and configure timeouts finegrained...`

Maybe could be `you have better control of how often Kafka Streams retries and can configure fine-grained timeouts..` ?

**Comment by mjsax** [comment]
Updated.

**Comment by mjsax** [comment]
Retest this please.

**Comment by guozhangwang** [comment]
LGTM.

**Pull request merged**